playwright/_impl/_frame.py

```diff
@@ -111,15 +111,15 @@ def __repr__(self) -> str:
     def _on_load_state(
         self, add: DocumentLoadState = None, remove: DocumentLoadState = None
     ) -> None:
+        if not self._parent_frame and add == "load" and self._page:
```

Comment:
This fixes [this](https://github.com/microsoft/playwright-python/blob/248f3ec434df48f3347aeecf9c0b0ae64b4e1714/tests/async/test_page.py#L167-L174) test because we used to dispatch the `load` event after we dispatched the `loadstate` update to the eventemitter.